iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.inl

```diff
@@ -84,26 +87,29 @@ inline constexpr Duration Duration::fromNanoseconds(const T value) noexcept
 template <typename T>
 inline constexpr Duration Duration::fromMicroseconds(const T value) noexcept
 {
-    auto clampedValue = positiveValueOrClampToZero(value);
-    auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MICROSECS_PER_SEC);
-    auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MICROSECS_PER_SEC)
-                                                            * Duration::NANOSECS_PER_MICROSEC);
+    const auto clampedValue = positiveValueOrClampToZero(value);
+    const auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MICROSECS_PER_SEC);
+    const auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MICROSECS_PER_SEC)
+                                                                  * Duration::NANOSECS_PER_MICROSEC);
     return Duration{seconds, nanoseconds};
 }
 template <typename T>
 inline constexpr Duration Duration::fromMilliseconds(const T value) noexcept
 {
-    auto clampedValue = positiveValueOrClampToZero(value);
-    auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MILLISECS_PER_SEC);
-    auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MILLISECS_PER_SEC)
-                                                            * Duration::NANOSECS_PER_MILLISEC);
+    const auto clampedValue = positiveValueOrClampToZero(value);
+    const auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MILLISECS_PER_SEC);
+    const auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MILLISECS_PER_SEC)
+                                                                  * Duration::NANOSECS_PER_MILLISEC);
     return Duration{seconds, nanoseconds};
 }
 template <typename T>
 inline constexpr Duration Duration::fromSeconds(const T value) noexcept
 {
-    auto clampedValue = positiveValueOrClampToZero(value);
+    const auto clampedValue = positiveValueOrClampToZero(value);
     constexpr Duration::Seconds_t MAX_SECONDS_BEFORE_OVERFLOW{std::numeric_limits<Duration::Seconds_t>::max()};
+
+    // AXIVION Next Construct AutosarC++19_03-A0.1.1 : Not dead code as it is platform dependent
```

Comment:
I don't see Axivion reporting this rule `A project shall not contain instances of non-volatile variables being
given values that are not subsequently used.` ~~Did you mean to suppress `M0.1.2`?~~ You did that below. I suppose this line can be deleted.